src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -403,65 +395,62 @@ private static Win32Program InternetShortcutProgram(string path)
                     {
                         urlPath = line.Substring(urlPrefix.Length);
 
-                        try
+                        if (!Uri.TryCreate(urlPath, UriKind.RelativeOrAbsolute, out Uri _))
                         {
-                            Uri uri = new Uri(urlPath);
-                        }
-                        catch (UriFormatException e)
-                        {
-                            // To catch the exception if the uri cannot be parsed.
-                            // Link to watson crash: https://watsonportal.microsoft.com/Failure?FailureSearchText=5f871ea7-e886-911f-1b31-131f63f6655b
-                            ProgramLogger.Exception($"url could not be parsed", e, MethodBase.GetCurrentMethod().DeclaringType, urlPath);
-                            return new Win32Program() { Valid = false, Enabled = false };
+                            ProgramLogger.Warn("url could not be parsed", null, MethodBase.GetCurrentMethod().DeclaringType, urlPath);
+                            return InvalidProgram;
                         }
 
                         // To filter out only those steam shortcuts which have 'run' or 'rungameid' as the hostname
-                        if (internetShortcutURLPrefixes.Match(urlPath).Success)
+                        if (InternetShortcutURLPrefixes.Match(urlPath).Success)
                         {
                             validApp = true;
                         }
                     }
-
-                    // Using OrdinalIgnoreCase since this is used internally
-                    if (line.StartsWith(iconFilePrefix, StringComparison.OrdinalIgnoreCase))
+                    else if (line.StartsWith(iconFilePrefix, StringComparison.OrdinalIgnoreCase))
                     {
                         iconPath = line.Substring(iconFilePrefix.Length);
                     }
+
+                    // If we resolved an urlPath & and an iconPath quit reading the file
+                    if (!string.IsNullOrEmpty(urlPath) && !string.IsNullOrEmpty(iconPath))
+                    {
+                        break;
+                    }
                 }
 
                 if (!validApp)
                 {
-                    return new Win32Program() { Valid = false, Enabled = false };
+                    return InvalidProgram;
                 }
 
                 try
                 {
-                    var p = new Win32Program
+                    return new Win32Program
                     {
                         Name = Path.GetFileNameWithoutExtension(path),
                         ExecutableName = Path.GetFileName(path),
                         IcoPath = iconPath,
-                        FullPath = urlPath,
+                        FullPath = urlPath.ToLowerInvariant(),
```

Comment:
All Other Win32Programs use LowerInvariant

Reply:
Can we change it ToUpperInvariant as it is recommended?

Reply:
Since its shown to the user (and all other types also use the lowercase variant) I don't think we should!